privacyidea/lib/smtpserver.py

```diff
@@ -113,6 +115,10 @@ def test_email(config, recipient, subject, body, sender=None,
             password = decryptPassword(config['password'])
             if password == FAILED_TO_DECRYPT_PASSWORD:
                 password = config['password']
+            # Under Python 2, we pass passwords as bytestrings to get CRAM-MD5 to work.
+            # Under Python 3, we pass passwords as unicode.
+            if six.PY2:
+                password = to_utf8(password)
```

Comment:
I can not tell you why, I feel slightly unpleasent with that.
Could we maybe add a config value, so that the password will *not* be converted, if the value is set - like

~~~~python
from privacyidea.lib.framework import get_app_config_value

if six.PY2 and get_app_config_value().get("PI_MAIL_UTF8_PASSWORDS", True):
    password = to_utf8(password)
~~~~
This way could "switch this off", by setting in pi.cfg

    PI_MAIL_UTF8_PASSWORDS = False

Reply:
I think always converting the password to a bytestring under Python 2 should be fine for all three supported authentication mechanisms PLAIN, LOGIN and CRAM-MD5. But OK, better be safe than sorry, I'll add the config option :) But I guess it's ok if we keep this option undocumented for now?

Reply:
BTW, just wanted to note that even with this change, passwords with *non-ASCII characters* (like ``passwörd``) won't work, because Python's smtplib doesn't really support them: https://bugs.python.org/issue29750.

Reply:
I added an option PI_SMTP_PASSWORD_AS_BYTES in c298946748f95ff7ede3f901d7cfdbbb0d9f600d. What do you think? :)

Reply:
If smtplib would one day support non-ASCII characters, to_utf8 would immediately work, while to_bytes would not. Right?